carma_wm/src/collision_detection.cpp

```diff
@@ -0,0 +1,179 @@
+#include "carma_wm/collision_detection.h"
+
+namespace carma_wm {
+
+    namespace collision_detection {
+
+        std::vector<cav_msgs::RoadwayObstacle> CollisionChecking::WorldCollisionDetection(cav_msgs::RoadwayObstacleList& rwol, cav_msgs::TrajectoryPlan& tp, geometry_msgs::Vector3& size, geometry_msgs::Twist& veloctiy, int target_time){
+
+            std::vector<cav_msgs::RoadwayObstacle> rwo_collison;
+
+            collision_detection::MovingObject vehicle_object = CollisionChecking::ConvertVehicleToMovingObject(tp, size, veloctiy);
+
+            for (int i = 0; i < rwol.roadway_obstacles.size(); i++) {
```

Comment:
best practice is to use size_t as the iterator type in for loops since it is unsigned.